Rubberduck.Core/UI/CodeExplorer/Commands/CodeExplorerExtractInterfaceCommand.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Refactorings.Exceptions;
+using Rubberduck.Refactorings.ExtractInterface;
+using Rubberduck.UI.Command.Refactorings.Notifiers;
+using Rubberduck.VBEditor.Events;
+using System;
+using System.Collections.Generic;
+
+namespace Rubberduck.UI.CodeExplorer.Commands
+{
+    public class CodeExplorerExtractInterfaceCommand : CodeExplorerCommandBase
+    {
+        private static readonly Type[] ApplicableNodes =
+        {
+            typeof(CodeExplorerComponentViewModel)
+        };
+
+        private readonly IParserStatusProvider _state;
+        private readonly ExtractInterfaceRefactoring _refactoring;
+        private readonly ExtractInterfaceFailedNotifier _failureNotifier;
+
+        public CodeExplorerExtractInterfaceCommand(
+            ExtractInterfaceRefactoring refactoring,
+            IParserStatusProvider state,
+            ExtractInterfaceFailedNotifier failureNotifier,
+            IVbeEvents vbeEvents) 
+            : base(vbeEvents)
+        {
+            _state = state;
+            _refactoring = refactoring;
+            _failureNotifier = failureNotifier;
+            AddToCanExecuteEvaluation(SpecialEvaluateCanExecute);
+            AddToOnExecuteEvaluation(FurtherCanExecuteEvaluation);
+        }
+
+        public sealed override IEnumerable<Type> ApplicableNodeTypes => ApplicableNodes;
+
+        private bool SpecialEvaluateCanExecute(object parameter)
+        {
+            return _state.Status == ParserState.Ready &&
+                   parameter is CodeExplorerComponentViewModel node &&
+                   _refactoring.CanExecute((RubberduckParserState)_state, node.QualifiedSelection.Value.QualifiedName);
```

Comment:
Instead of casting, I'd rather that it used the interface or modify the class to depend on `RubberduckParserState` instead of the interface.

Reply:
Yes, please do not cast an injected interface to an implementation. Which implementation we inject may change at any time and such a change is not expected to break anything.